### DISCUSSION TITLE
Installation Assistance

### DISCUSSION THREAD

**Discussion by knewmania** [jasonacox/Powerwall-Dashboard#116] [Q&A] [answered]
Hello all. 
I am attempting to install the Powerwall-Dashboard on a RPi v4 Ubuntu 22.04.1 LTS and am running into some issues.

I have installed Docker and when I issue 'sudo docker --version' I get "Docker version 20.10.21, build baeda1f".
I have done the git clone and no errors were presented.
When I run the ./setup.sh script after I enter the IP Address for my Powerwall Gateway I receive the following error:

"./setup.sh: line 91: pypowerwall.env: Permission denied"

I have run sudo usermod -aG docker $USER and rebooted, but I get the same results.
I have confirmed that my user (pi) is a member of the docker group 

grep /etc/group -e "docker"
docker:x:999:pi

Any assistance would be appreciated. Thanks for your time.

Knewmania

**Comment by knewmania** [comment, marked as answer]
I resolved the issue. Looks like I had failed to install Docker-Compose.

**Reply by jasonacox** [reply]
Hi @knewmania !  Thanks for opening this!  That's interesting.  The error you received was not related to docker-compose.  That line 91 is attempting to write a local file to your local directory (pypowerwall.env) which holds the data you entered during Setup and is used by pypowerwall to connect to your Tesla system.  Is it possible that the user (pi?) did not have writes to the directory you were in?  It sounds like the second time you ran it, it worked, but it is curious.  I could add some logic to the `setup.sh` script to ensure the user you are using has access to write to the local directory before proceeding. 

As to the docker-compose.  What did you do to install docker-compose?

The https://github.com/jasonacox/Powerwall-Dashboard/blob/main/compose-dash.sh script should have detected that you were missing Docker-Compose.  @BuongiornoTexas added logic in there to detect for `docker-compose` or `docker compose` but perhaps it is not firing correctly?

**Reply by knewmania** [reply]
I guess it is possible that the pi account did not have permissions to the folder/files. I did see some access denied entries in some of the logs. I ran through the install multiple times and encountered the same issue multiple times (re-baselining the Pi SD card). The only problem I could see was that docker-compose was not installed.

I ran 'sudo apt install docker-compose' in order to install docker-compose.

I am by no means an expert in any of this, so it's very possible I just did something wrong. I did my best to follow the instructions as they were written.

Let me know if you need anything.

**Reply by jasonacox** [reply]
Thank you @knewmania !  This feedback is great!  I'm always wanting to make it easier for people to install and get the dashboard running.  Speaking of which, do you have a working Dashboard at this point?  I assumed so, but want to verify. 

I think I have discovered a missing step in our "stey by step" instructions - I wonder if this would have been your case:

```bash
git clone https://github.com/jasonacox/Powerwall-Dashboard.git
./setup.sh
```

That should include instruction to move into the folder that was downloaded (`cd`).  Also, it assumes you are in a directory that you own (e.g. `cd ~`).

```bash
git clone https://github.com/jasonacox/Powerwall-Dashboard.git
cd Powerwall-Dashboard
./setup.sh
```

**Reply by knewmania** [reply]
Yes I have the Dashboard functioning now, and I love it.

I would say that if there is anyway you could verify some of the prerequisites during the install and provide feedback if a component is missing that would be helpful. Not sure why my install did not note that, if there is already logic in place to do that. Maybe output the versions of the required components as part of the install (docker-compose --version), and if the version is null, then error with a message saying a prerequisite is missing? Also the issue where the current user has to be a member of the docker group. You could either check for that, or just run the sudo usermod -aG docker $USER as part of the install?

**Reply by BJReplay** [reply]
> Thank you @knewmania ! This feedback is great! I'm always wanting to make it easier for people to install and get the dashboard running. Speaking of which, do you have a working Dashboard at this point? I assumed so, but want to verify.
> 
> I think I have discovered a missing step in our "stey by step" instructions - I wonder if this would have been your case:
> 
> ```shell
> git clone https://github.com/jasonacox/Powerwall-Dashboard.git
> ./setup.sh
> ```
> 
> That should include instruction to move into the folder that was downloaded (`cd`). Also, it assumes you are in a directory that you own (e.g. `cd ~`).
> 
> ```shell
> git clone https://github.com/jasonacox/Powerwall-Dashboard.git
> cd Powerwall-Dashboard
> ./setup.sh
> ```

That was definitely a missing instruction.  It's not that hard to figure out if you know a little bit about what you're doing, but did cause me to stop for a second and check that I was following the instructions.  A quick ls and I figured out the step that I needed to insert.
